diff --git a/libswc/shell_surface.c b/libswc/shell_surface.c
index 25d4e39..0d4d96a 100644
--- a/libswc/shell_surface.c
+++ b/libswc/shell_surface.c
@@ -190,12 +190,12 @@ static void set_title(struct wl_client * client, struct wl_resource * resource,
     window_set_title(&shell_surface->window, title, -1);
 }
 
-static void set_class(struct wl_client * client, struct wl_resource * resource,
-                      const char * class)
+static void set_appid(struct wl_client * client, struct wl_resource * resource,
+                      const char * appid)
 {
     struct shell_surface * shell_surface = wl_resource_get_user_data(resource);
 
-    window_set_class(&shell_surface->window, class);
+    window_set_appid(&shell_surface->window, appid);
 }
 
 static const struct wl_shell_surface_interface shell_surface_implementation = {
@@ -208,7 +208,7 @@ static const struct wl_shell_surface_interface shell_surface_implementation = {
     .set_popup = &set_popup,
     .set_maximized = &set_maximized,
     .set_title = &set_title,
-    .set_class = &set_class
+    .set_class = &set_appid
 };
 
 static void handle_surface_destroy(struct wl_listener * listener, void * data)
diff --git a/libswc/swc.h b/libswc/swc.h
index 22d2945..965720a 100644
--- a/libswc/swc.h
+++ b/libswc/swc.h
@@ -108,9 +108,9 @@ struct swc_window_handler
     void (* title_changed)(void * data);
 
     /**
-     * Called when the window's class changes.
+     * Called when the window's app_id changes.
      */
-    void (* class_changed)(void * data);
+    void (* appid_changed)(void * data);
 
     /**
      * Called when the window's parent changes.
@@ -147,7 +147,7 @@ struct swc_window_handler
 struct swc_window
 {
     char * title;
-    char * class;
+    char * app_id;
 
     struct swc_window * parent;
 };
diff --git a/libswc/window.c b/libswc/window.c
index d035f07..03033a3 100644
--- a/libswc/window.c
+++ b/libswc/window.c
@@ -389,7 +389,7 @@ bool window_initialize(struct window * window, const struct window_impl * impl,
     DEBUG("Initializing window, %p\n", window);
 
     window->base.title = NULL;
-    window->base.class = NULL;
+    window->base.app_id = NULL;
     window->base.parent = NULL;
 
     if (!(window->view = swc_compositor_create_view(surface)))
@@ -428,7 +428,7 @@ void window_finalize(struct window * window)
     window_unmanage(window);
     compositor_view_destroy(window->view);
     free(window->base.title);
-    free(window->base.class);
+    free(window->base.app_id);
 }
 
 void window_manage(struct window * window)
@@ -460,13 +460,13 @@ void window_set_title(struct window * window, const char * title, size_t length)
         window->handler->title_changed(window->handler_data);
 }
 
-void window_set_class(struct window * window, const char * class)
+void window_set_appid(struct window * window, const char * app_id)
 {
-    free(window->base.class);
-    window->base.class = strdup(class);
+    free(window->base.app_id);
+    window->base.app_id = strdup(app_id);
 
-    if (window->handler->class_changed)
-        window->handler->class_changed(window->handler_data);
+    if (window->handler->appid_changed)
+        window->handler->appid_changed(window->handler_data);
 }
 
 void window_set_parent(struct window * window, struct window * parent)
diff --git a/libswc/window.h b/libswc/window.h
index b5ac1a7..e8bd896 100644
--- a/libswc/window.h
+++ b/libswc/window.h
@@ -103,7 +103,7 @@ void window_unmanage(struct window * window);
 void window_set_title(struct window * window,
                       const char * title, size_t length);
 
-void window_set_class(struct window * window, const char * class);
+void window_set_appid(struct window * window, const char * app_id);
 
 void window_set_parent(struct window * window, struct window * parent);
 
diff --git a/libswc/xdg_surface.c b/libswc/xdg_surface.c
index 5097aba..3c09fa3 100644
--- a/libswc/xdg_surface.c
+++ b/libswc/xdg_surface.c
@@ -186,7 +186,7 @@ static void set_app_id(struct wl_client * client, struct wl_resource * resource,
 {
     struct xdg_surface * surface = wl_resource_get_user_data(resource);
 
-    window_set_class(&surface->window, app_id);
+    window_set_appid(&surface->window, app_id);
 }
 
 static void show_window_menu(struct wl_client * client,
